Cython/Compiler/Symtab.py

```diff
@@ -2314,6 +2314,25 @@ def needs_tp_clear(self):
         """
         return self.needs_gc() and not self.directives.get('no_gc_clear', False)
 
+    def may_have_finalize(self):
+        """
+        This covers cases where we definitely have a __del__ function
+        and also cases where one of the base classes could have a __del__
+        function but we don't know.
+        """
+        current_type_scope = self
+        while current_type_scope:
+            del_entry = current_type_scope.lookup_here("__del__")
+            if del_entry and del_entry.is_special:
```

Comment:
Wouldn't a simple `self.lookup()` also work here? It should look properly through the hierarchy.

I understand that there's also the "can't tell for sure" case below, so this implementation might be a good one. I'd just like to raise the question here to make sure it's been considered.

Reply:
I believe `__del__` doesn't appear in `lookup` because it's a def function and only cdef functions get copied into derived classes scopes. I did try `self.lookup` because it'd have been a lot easier if it had worked.